README.md
### Twitter Clone

A twitter clone written in SwiftUI


CHANGELOG.md

### CHANGELOG

2 September 2022:
- Home Screen complete
- Profile Screen complete

30 Aug 2022:
- Profile View header complete
- Profile View bio complete
- Profile View tab switcher complete 

12 Aug 2022:
- Initial commit


TwitterClone/TweetView.swift
//
//  TweetView.swift
//  TwitterClone
//
//  Created by Muhd Mirza on 31/8/22.
//

import SwiftUI

struct TweetView: View {
    var body: some View {
        HStack(alignment: .top) {
            Image("lakemcdonald").resizable().frame(width: 50, height: 50).clipShape(Circle())
            
            VStack(alignment: .leading) {
                HStack {
                    Text("mirza").font(Font.subheadline.weight(.heavy))
                    Text("@muhdmirzamz")
                    Text(".")
                    Text("3d")
                    
                    Spacer()
                    
                    Image(systemName: "ellipsis")
                }
                .foregroundColor(Color.white)
                
                Text("This is a test tweet. This will test the flexibility and line wrapping etc. Hope it looks good. Testing testing testing.")
                    .foregroundColor(Color.white)
                
                HStack(spacing: 50) {
                    Image(systemName: "bubble.middle.bottom")
                    Image(systemName: "arrow.rectanglepath")
                    Image(systemName: "heart")
                    Image(systemName: "square.and.arrow.up")
                }
                .padding(.top, 2)
                .foregroundColor(Color.white)
            }
        }
    }
}

struct TweetView_Previews: PreviewProvider {
    static var previews: some View {
        TweetView()
            .background(Color.init(red: 0/255, green: 26/255, blue: 51/255))
    }
}


TwitterClone/ProfileView.swift
//
//  ProfileView.swift
//  TwitterClone
//
//  Created by Muhd Mirza on 22/8/22.
//

import SwiftUI

struct ProfileView: View {
    var body: some View {
        NavigationView {
            VStack {
                HeaderView()
                    .ignoresSafeArea(edges: .top)
                    .offset(y: -100)
                    .padding(.bottom, -100)
                    

                HStack {
                    Image("icybay")
                        .resizable()
                        .frame(width: 100, height: 100)
                        .clipShape(Circle())

                    Spacer()

                    Button.init {

                    } label: {
                        Text("Edit Profile")
                    }
                    .frame(width: 120)
                    .foregroundColor(Color.white)
                    .padding(2)
                    .overlay(
                        // apply an overlay here for smooth rounded corners
                        RoundedRectangle(cornerRadius: 20)
                            .stroke(.white, lineWidth: 1)
                    )

                }
                .padding()
                .offset(y: -50)
                .padding(.bottom, -50)
//                .background(Color.red)

                ProfileInfoView()
                    .offset(y: -30)
                    .padding(.bottom, -30)
//                    .background(Color.red)

                ProfileTabView()

                ScrollView {
                    ForEach(0..<20) { tweet in
                        TweetView()
                            .padding(.leading, 5)
                            .padding()
                    }
                }
                
//                Spacer()
            }
            .background(Color.init(red: 0/255, green: 26/255, blue: 51/255))
            .navigationBarHidden(true)
        }
    }
}

struct ProfileView_Previews: PreviewProvider {
    static var previews: some View {
        ProfileView()
    }
}


TwitterClone/ProfileTabView.swift
//
//  ProfileTabView.swift
//  TwitterClone
//
//  Created by Muhd Mirza on 30/8/22.
//

import SwiftUI

struct ProfileTabView: View {
    var body: some View {
        VStack {
            HStack() {
                Text("Tweets")
                    .font(Font.headline.weight(.heavy))
                Spacer()
                Text("Tweets & replies")
                Spacer()
                Text("Media")
                Spacer()
                Text("Likes")
            }
            .foregroundColor(Color.white)
            .padding(.leading, 25)
            .padding(.top, 15)
            .padding(.trailing, 25)
            .background(Color.init(red: 0/255, green: 26/255, blue: 51/255))

            Divider().background(Color.white)
        }
    }
}

struct ProfileTabView_Previews: PreviewProvider {
    static var previews: some View {
        ProfileTabView()
    }
}


TwitterClone/ProfileInfoView.swift
//
//  ProfileInfoView.swift
//  TwitterClone
//
//  Created by Muhd Mirza on 29/8/22.
//

import SwiftUI

struct ProfileInfoView: View {
    var body: some View {
        VStack(alignment: .leading) {
            VStack(alignment: .leading) {
                Text("mirza")
                    .font(Font.headline.weight(.heavy))
                    .foregroundColor(Color.white)
                Text("@muhdmirzamz")
                    .foregroundColor(Color.white)
            }
            .padding(.bottom, 20)
            
            Text("Frontend software engineeer dabbling with iOS on the side. Exploring design with @fof_singapore and building communities @msociety_dev")
                .lineLimit(nil)
                .frame(height: 100)
                .foregroundColor(Color.white)
            
            HStack {
                Text("Singapore")
                Text("muhdmirzamz.github.io")
                    .padding(.leading, 50)
            }
            .foregroundColor(Color.white)
            .padding(.top, 10)
            
            Text("Joined September 2013")
                .foregroundColor(Color.white)
                .padding(.top, 10)

            
            HStack {
                Text("102")
                    .font(Font.headline.weight(.heavy))
                    .foregroundColor(Color.white)
                Text("Followers")
                    .foregroundColor(Color.white)
                Text("72")
                    .font(Font.headline.weight(.heavy))
                    .foregroundColor(Color.white)
                Text("Followers")
                    .foregroundColor(Color.white)
            }.padding(.top, 10)
        }
        .padding()
    }
}

struct ProfileInfoView_Previews: PreviewProvider {
    static var previews: some View {
        ProfileInfoView()
            .background(Color.red)
    }
}


TwitterClone/HomeView.swift
//
//  HomeView.swift
//  TwitterClone
//
//  Created by Muhd Mirza on 30/8/22.
//

import SwiftUI

struct HomeView: View {
    
    init() {
        UINavigationBar.appearance().titleTextAttributes = [.foregroundColor: UIColor.white]
    }
    
    var body: some View {
        
        NavigationView {
            ZStack {
                VStack {
                    ScrollView {
                        ForEach(0..<20) { tweet in
                            TweetView()
                                .padding(.leading, 5)
                                .padding()
                        }
                    }
                }
                .navigationTitle("Twitter")
                .navigationBarTitleDisplayMode(.inline)
                .background(Color.init(red: 0/255, green: 26/255, blue: 51/255))

            }
            .background(Color.init(red: 0/255, green: 26/255, blue: 51/255))
            
        }
    }
}

struct HomeView_Previews: PreviewProvider {
    static var previews: some View {
        HomeView()
    }
}
